packages/next/build/swc/src/styled_jsx/transform_css.rs

```diff
@@ -0,0 +1,256 @@
+use easy_error::{bail, Error};
+use swc_common::{source_map::Pos, BytePos, Span, SyntaxContext, DUMMY_SP};
+use swc_css::ast::*;
+use swc_css::parser::{parse_str, parse_tokens, parser::ParserConfig};
+use swc_css::visit::{VisitMut, VisitMutWith};
+use swc_css_codegen::{
+  writer::basic::{BasicCssWriter, BasicCssWriterConfig},
+  CodegenConfig, Emit,
+};
+use swc_ecmascript::ast::{Expr, Str, StrKind, Tpl, TplElement};
+use swc_ecmascript::utils::HANDLER;
+use swc_stylis::prefixer::prefixer;
+
+use super::{hash_string, string_literal_expr, LocalStyle};
+
+pub fn transform_css(
+  style_info: &LocalStyle,
+  is_global: bool,
+  class_name: &Option<String>,
+) -> Result<Expr, Error> {
+  let result: Result<Stylesheet, _> = parse_str(
+    &style_info.css,
+    style_info.css_span.lo,
+    style_info.css_span.hi,
+    ParserConfig {
+      parse_values: false,
+    },
+  );
+  let mut ss = match result {
+    Ok(ss) => ss,
+    Err(_) => {
+      HANDLER.with(|handler| {
+        handler
+          .struct_span_err(
+            style_info.css_span,
+            "Failed to parse css in styled jsx component",
+          )
+          .emit()
+      });
+      bail!("Failed to parse css");
+    }
+  };
+  // ? Do we need to support optionally prefixing?
+  ss.visit_mut_with(&mut prefixer());
+  ss.visit_mut_with(&mut Namespacer {
```

Comment:
will it also support to bail out no nesting css? [ref](https://github.com/vercel/styled-jsx/blob/8b585d5907add2228f3651225de80f99dc069a54/src/lib/style-transform.js#L6-L19)

Reply:
Currently, nested css is not supported because it's not a standard.
But I can implement it if required.

Reply:
sgtm If it can fail on parsing 👍 we don't have plan to support nested css yest